src/replica/bulk_load/replica_bulk_loader.cpp

```diff
@@ -403,80 +387,114 @@ error_code replica_bulk_loader::download_sst_files(const std::string &remote_dir
         return ERR_FILE_OPERATION_FAILED;
     }
 
+    // start download
+    _is_downloading.store(true);
+    _download_task = tasking::enqueue(
+        LPC_BACKGROUND_BULK_LOAD,
+        tracker(),
+        std::bind(
+            &replica_bulk_loader::download_files, this, provider_name, remote_dir, local_dir));
+    return ERR_OK;
+}
+
+// ThreadPool: THREAD_POOL_DEFAULT
+void replica_bulk_loader::download_files(const std::string &provider_name,
+                                         const std::string &remote_dir,
+                                         const std::string &local_dir)
+{
+    FAIL_POINT_INJECT_F("replica_bulk_loader_download_files", [](string_view) {});
+
+    ddebug_replica("start to download files");
     dist::block_service::block_filesystem *fs =
         _stub->_block_service_manager.get_or_create_block_filesystem(provider_name);
 
-    // download metadata file synchronously
-    uint64_t file_size = 0;
-    error_code err = _stub->_block_service_manager.download_file(
-        remote_dir, local_dir, bulk_load_constant::BULK_LOAD_METADATA, fs, file_size);
-    if (err != ERR_OK && err != ERR_PATH_ALREADY_EXIST) {
-        derror_replica("download bulk load metadata file failed, error = {}", err.to_string());
-        return err;
-    }
+    {
+        zauto_write_lock l(_lock);
+        // download metadata file synchronously
+        uint64_t file_size = 0;
+        error_code err = _stub->_block_service_manager.download_file(
+            remote_dir, local_dir, bulk_load_constant::BULK_LOAD_METADATA, fs, file_size);
```

Comment:
```
error_code err = _stub->_block_service_manager.download_file(
            remote_dir, local_dir, bulk_load_constant::BULK_LOAD_METADATA, fs, file_size);
```
does these code need the protection of _lock?

Reply:
They are needn't to be locked, I have updated it.